phoenix-core/src/it/java/org/apache/phoenix/end2end/LogicalTableNameExtendedIT.java

```diff
@@ -0,0 +1,281 @@
+        String view2IndexName1 = "VW2IDX1_" + generateUniqueName();
+
+         try (Connection conn = getConnection(propsNamespace)) {
+            try (Connection conn2 = getConnection(propsNamespace)) {
+                conn.createStatement().execute("CREATE SCHEMA IF NOT EXISTS " + schemaName);
+                testWithViewsAndIndex_BaseTableChange(conn, conn2, null, schemaName, tableName, view1Name,
+                        view1IndexName1, view1IndexName2, view2Name, view2IndexName1, true, false);
+
+                populateView(conn, (schemaName+"."+view2Name), 10, 1);
+                ResultSet rs = conn2.createStatement().executeQuery("SELECT * FROM " + (schemaName + "." + view2IndexName1) + " WHERE \":PK1\"='PK10'");
+                assertEquals(true, rs.next());
+
+            }
+        }
+    }
+
+    private void test_bothTableAndIndexHaveDifferentNames(Connection conn, Connection conn2, String schemaName, String tableName, String indexName) throws Exception {
+        String fullTableHName = schemaName + ":" + tableName;
+        String fullIndexHName = schemaName + ":" + indexName;
+
+        conn.createStatement().execute("CREATE SCHEMA IF NOT EXISTS " + schemaName);
+        // Create tables and change physical index table
+        test_IndexTableChange(conn, conn2, schemaName, tableName, indexName,
+                IndexRegionObserver.UNVERIFIED_BYTES, true);
+        // Now change physical data table
+        createAndPointToNewPhysicalTable(conn, fullTableHName, true);
+        try (HBaseAdmin admin = conn.unwrap(PhoenixConnection.class).getQueryServices()
+                .getAdmin()) {
+            assertEquals(false, admin.tableExists(TableName.valueOf(fullTableHName)));
+            assertEquals(false, admin.tableExists(TableName.valueOf(fullIndexHName)));
+        }
+    }
+
+    @Test
+    public void testUpdatePhysicalTableName_bothTableAndIndexHaveDifferentNames() throws Exception {
+        String schemaName = "S_" + generateUniqueName();
+        String tableName = "TBL_" + generateUniqueName();
+        String indexName = "IDX_" + generateUniqueName();
+        String fullTableName = SchemaUtil.getTableName(schemaName, tableName);
+        String fullTableHName = schemaName + ":" + tableName;
+        String fullIndexName = SchemaUtil.getTableName(schemaName, indexName);
+        String fullIndexHName = schemaName + ":" + indexName;
+        String fullNewTableHName = schemaName + ":NEW_TBL_" + tableName;
+        try (Connection conn = getConnection(propsNamespace)) {
+            try (Connection conn2 = getConnection(propsNamespace)) {
+                test_bothTableAndIndexHaveDifferentNames(conn, conn2, schemaName, tableName, indexName);
+                try (HBaseAdmin admin = conn.unwrap(PhoenixConnection.class).getQueryServices()
+                        .getAdmin()) {
+
+                    assertEquals(false, admin.tableExists(TableName.valueOf(fullTableHName)));
```

Comment:
Aren't you already doing this check in test_bothTableAndIndexHaveDifferentNames which you just called?

Reply:
You are right. Removing